client/my-sites/site-settings/controller.js

```diff
@@ -150,6 +151,12 @@ export function legacyRedirects( context, next ) {
 		'connected-apps': '/me/security/connected-applications',
 	};
 
+	if ( section === 'account' && context.query.new_email_result ) {
+		return page.redirect(
+			addQueryArgs( '/me/account', { new_email_result: context.query.new_email_result } )
+		);
+	}
```

Comment:
This is the legacyRedirects middleware. Currently a confirmed email change redirects to a legacy route which is then re-routed to /me/account. Without this change, it would lose its query param noting that the new email change was a success.